frontend/src/SelectLanguage.js

```diff
@@ -0,0 +1,97 @@
+import { Fragment, useContext, useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import { LanguageContext } from "./Contexts/Contexts";
+import axios from "axios";
+import "bootstrap-icons/font/bootstrap-icons.css";
+
+const SelectLanguage = () => {
+  const [languages, setLanguages] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const { language, setLanguage } = useContext(LanguageContext);
+
+  useEffect(() => {
+    axios
+      .get("http://127.0.0.1:8000/languages/")
+      .then((response) => {
+        setLanguages(response.data);
+      })
+      .catch((error) => {
+        console.error("Error fetching data", error);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
+
+  const handleRadioChange = (event) => {
+    setLanguage(event.target.value);
+  };
+
+  if (loading) {
+    return (
+      <>
+        <div className="container d-flex justify-content-center align-items-center min-vh-100">
```

Comment:
You might be able to implement this as a loop going through an array for different `text-` classes.